src/java/io/bazel/rulesscala/worker/Worker.java

```diff
@@ -81,7 +81,14 @@ public void checkPermission(Permission permission) {
           int code = 0;
 
           try {
-            workerInterface.work(stringListToArray(request.getArgumentsList()));
+            String[] workerArgs = stringListToArray(request.getArgumentsList());
+            String[] args;
+            if (workerArgs.length == 1) {
```

Comment:
can we not unify this part as well? Seems it is duplicated below.

Reply:
Thanks for the catch!